# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/tar/tar-bzip2.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/tar.c.orig	2004-04-04 11:53:30.000000000 +0200
+++ ./src/tar.c	2004-06-24 13:48:06.757203696 +0200
@@ -239,6 +239,8 @@
   {"block-number", no_argument, 0, 'R'},
   {"blocking-factor", required_argument, 0, 'b'},
   {"bzip2", no_argument, 0, 'j'},
+  {"bzip2", no_argument, 0, 'y'},
+  {"bzip2", no_argument, 0, 'I'},
   {"catenate", no_argument, 0, 'A'},
   {"checkpoint", no_argument, 0, CHECKPOINT_OPTION},
   {"check-links", no_argument, &check_links_option, 1},
@@ -458,7 +460,7 @@
                                      control pax keywords\n\
   -V, --label=NAME                   create archive with volume name NAME\n\
               PATTERN                at list/extract time, a globbing PATTERN\n\
-  -j, --bzip2                        filter the archive through bzip2\n\
+  -j, -y, -I, --bzip2                filter the archive through bzip2\n\
   -z, --gzip, --ungzip               filter the archive through gzip\n\
   -Z, --compress, --uncompress       filter the archive through compress\n\
       --use-compress-program=PROG    filter through PROG (must accept -d)\n"),
@@ -768,13 +770,15 @@
 	ignore_zeros_option = true;
 	break;
 
-      case 'I':
-	USAGE_ERROR ((0, 0,
-		      _("Warning: the -I option is not supported;"
-			" perhaps you meant -j or -T?")));
+      case 'j':
+	set_use_compress_program_option ("bzip2");
 	break;
 
-      case 'j':
+      case 'y':
+	set_use_compress_program_option ("bzip2");
+	break;
+
+      case 'I':
 	set_use_compress_program_option ("bzip2");
 	break;
 
@@ -943,12 +947,6 @@
 	  }
 	break;
 
-      case 'y':
-	USAGE_ERROR ((0, 0,
-		      _("Warning: the -y option is not supported;"
-			" perhaps you meant -j?")));
-	break;
-
       case 'z':
 	set_use_compress_program_option ("gzip");
 	break;
